test/IdRegistry/IdRegistry.symbolic.t.sol

```diff
@@ -0,0 +1,183 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.19;
+
+import {SymTest} from "halmos-cheatcodes/SymTest.sol";
+import {Test} from "forge-std/Test.sol";
+
+import {IdRegistryHarness} from "../Utils.sol";
+
+contract IdRegistrySymTest is SymTest, Test {
+    IdRegistryHarness idRegistry;
+    address trustedCaller;
+    address x;
+    address y;
+
+    function setUp() public {
+        // Setup IdRegistry
+        idRegistry = new IdRegistryHarness(address(this));
+
+        trustedCaller = address(0x1000);
+        idRegistry.setTrustedCaller(trustedCaller);
+
+        // Register fids
+        vm.prank(trustedCaller);
+        idRegistry.trustedRegister(address(0x1001), address(0x2001));
+        vm.prank(trustedCaller);
+        idRegistry.trustedRegister(address(0x1002), address(0x2002));
+
+        assert(idRegistry.idOf(address(0x1001)) == 1);
+        assert(idRegistry.idOf(address(0x1002)) == 2);
+
+        assert(idRegistry.getRecoveryOf(1) == address(0x2001));
+        assert(idRegistry.getRecoveryOf(2) == address(0x2002));
+
+        // Create symbolic addresses
+        x = svm.createAddress("x");
+        y = svm.createAddress("y");
+    }
+
+    function verifyInvariants(bytes4 selector, address caller) public {
```

Comment:
note that the default prefix for halmos test functions is `check_`. if you want to use a different prefix, then you need to provide an additional option `--function <prefix>` to halmos, which is `--function verify` in this case. otherwise these functions will be omitted from testing, as you can see in the last ci run:
https://github.com/farcasterxyz/contracts/actions/runs/5671611054/job/15369109292